libqtile/core/manager.py

```diff
@@ -312,6 +311,12 @@ def _process_screens(self) -> None:
             scr._configure(self, i, x, y, w, h, grp)
             screens.append(scr)
 
+        for screen in self.screens:
+            if screen not in screens:
+                for gap in screen.gaps:
+                    if isinstance(gap, bar.Bar) and gap.window:
+                        gap.kill_window()
+
```

Comment:
Isn't ref counting supposed to remove the screens and associated bar if a screen is not in use anymore? Do we still hold a reference to unused screens, and if so, why?

Reply:
No because the screens and bars persist within `qtile.config` and so rather than being made and destroyed they are applied or removes to physical outputs. The `Internal` window itself however can be destroyed and recreated.

Reply:
Shouldn't we destroy stuff and reread config when appropriate?

Reply:
I don't think so - and this would be a significant change in itself. Think about how one might configure two `Screen`s in their config because they occasionally add a second screen. The second one waits around some of the time, ready to be used, rather than us deleting it right after `_process_screens` if there is only one screen. If we wanted to go down the route of trimming any unnecessary memory usage from `qtile.config` (which might be a good idea eventually), this would require the config-reloading mechanism to be solid and trusted as there are many factors that need to be taken into account (one example being preventing hooks from being subscribed more than once).

Reply:
We're already protected against duplicate hooks, IIRC. :P

Anyway, I see your point. I still think we shouldn't keep references for stuff we don't use, but practicality beats purity so let's move on!